ci-operator/step-registry/optional-operators/cvp-common/scorecard/optional-operators-cvp-common-scorecard-commands.sh

```diff
@@ -9,19 +9,20 @@ set -x
 # or retries_max is reached.
 # Takes 2 arguments - the command/function and the output file
 run_scorecard() {
-  local retries_max=3
-  local attempt_num=1
-  until [ -s "$2" ]
-
-  do
-    if (( attempt_num>retries_max )); then
-      echo "All $retries_max attempts of scorecard tests failed. No more attempts. "
-      return 1
-    else
-      echo "Attempt $attempt_num of $retries_max."
-      sleep 5
-      ((attempt_num++))
-    fi
+  
+  # try to the run tests for the first time to generate the file
+  "$1" "$2"
+  local max_retries=6
+  # Define a counter for the number of retries
+  local retries=1
+  search_string="timed out waiting for the condition"
+  # Keep searching for the string in the file until it is not found
+  while grep -q "$search_string" "$2" && [ $retries -ne $max_retries ]; do
+      echo "Checking if there is a timeout string: $search_string"
```

Comment:
You already checked "timeout string" here and was missing.
The message is confusing.